### ISSUE TITLE
Errata syncing in repo-sync throws an error for newly added distribution openEuler 22.03

### ISSUE THREAD

**Issue by raulillo82** [uyuni-project/uyuni#5617]
# Problem description

Running a spacewalk-repo-sync of the new channel corresponding to the updates of openEuler fails. If errata is excluded in the options (--no-errata), reposync works correctly

Related PR is #5611

## Version of Uyuni Server and Proxy (if used)

```
uyuni:~ # zypper info Uyuni-Server-release
Retrieving repository 'Update repository of openSUSE Backports' metadata ..................................................................................................................................[done]
Building repository 'Update repository of openSUSE Backports' cache .......................................................................................................................................[done]
Loading repository data...
Reading installed packages...


Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : uyuni-server-devel
Name           : Uyuni-Server-release
Version        : 2022.06-183.9.uyuni2
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : Uyuni-Server-release-2022.06-183.9.uyuni2.src
Summary        : Uyuni Server
Description    : 
    Uyuni lets you efficiently manage physical, virtual,
    and cloud-based Linux systems. It provides automated and cost-effective
    configuration and software management, asset management, and system
    provisioning.

uyuni:~ # zypper info Uyuni-Proxy-release
Loading repository data...
Reading installed packages...


package 'Uyuni-Proxy-release' not found.

```

## Details about the issue

```
uyuni:~ # spacewalk-repo-sync -c openeuler2203-update-x86_64 
13:45:32 ======================================
13:45:32 | Channel: openeuler2203-update-x86_64
13:45:32 ======================================
13:45:32 Sync of channel started.
Retrieving repository 'openeuler2203-update-x86_64' metadata ..............................................................................................................................................[done]
Building repository 'openeuler2203-update-x86_64' cache ...................................................................................................................................................[done]
All repositories have been refreshed.
13:45:41 Repo URL: https://repo.openeuler.org/openEuler-22.03-LTS/update/x86_64?<token>
13:45:41     Packages in repo:              1283
13:45:43     No new packages to sync.
13:45:43 
13:45:43   Patches in repo: 105.
13:45:43 Unexpected error: <class 'TypeError'>
13:45:43 Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 620, in sync
    self.import_updates(plugin)
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 763, in import_updates
    processed_updates_count = self.upload_updates(notices)
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 966, in upload_updates
    notice = self.fix_notice(notice)
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 1541, in fix_notice
    if "." in notice['version']:
TypeError: argument of type 'NoneType' is not iterable

13:45:43 Total time: 0:00:10
```

**Comment by juliogonzalez** [comment]
@meaksh maybe you can have a look at some point? Other than that , reposync is working just fine :-)

**Comment by vzhestkov** [comment]
@raulillo82 do you have the test system deployed still? Could you please check the output of `spacewalk-repo-sync -f -c openeuler2203-update-x86_6`?

**Comment by raulillo82** [comment]
> @raulillo82 do you have the test system deployed still? Could you please check the output of `spacewalk-repo-sync -f -c openeuler2203-update-x86_6`?

Sure. Actually you can probably even access the system, as it's in a network to which you have access through vpn (ping me in that case for details). See details below:

```
vm-cachitomio:~ # export LANG=C && spacewalk-repo-sync -f -c openeuler2203-update-x86_64
18:20:20 ======================================
18:20:20 | Channel: openeuler2203-update-x86_64
18:20:20 ======================================
18:20:20 Sync of channel started.
Retrieving repository 'openeuler2203-update-x86_64' metadata .....................................................................................................................................[done]
Building repository 'openeuler2203-update-x86_64' cache ..........................................................................................................................................[done]
All repositories have been refreshed.
18:20:30 Repo URL: https://repo.openeuler.org/openEuler-22.03-LTS/update/x86_64?<token>
18:20:30     Packages in repo:              1491
18:20:44     Packages already synced:       1351
18:20:44     Packages to sync:               140
18:20:44     New packages to download:       140
18:20:44   Downloading packages:
18:20:44 Downloading total 140 files from 1 queues.
18:20:49     1/140 : atune-client-1.0.0-4.oe2203.x86_64.rpm
18:20:50     2/140 : bpftool-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:20:51     3/140 : bpftool-debuginfo-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:20:53     4/140 : bpftool-debuginfo-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:20:53     5/140 : bsdcat-3.5.2-4.oe2203.x86_64.rpm
18:20:53     6/140 : atune-db-1.0.0-4.oe2203.x86_64.rpm
18:20:54     7/140 : bsdcpio-3.5.2-4.oe2203.x86_64.rpm
18:20:54     8/140 : clibcni-2.0.7-2.oe2203.x86_64.rpm
18:20:54     9/140 : bsdtar-3.5.2-4.oe2203.x86_64.rpm
18:20:55     10/140 : clibcni-debuginfo-2.0.7-2.oe2203.x86_64.rpm
18:20:55     11/140 : clibcni-debugsource-2.0.7-2.oe2203.x86_64.rpm
18:20:55     12/140 : clibcni-devel-2.0.7-2.oe2203.x86_64.rpm
18:20:55     13/140 : curl-7.79.1-7.oe2203.x86_64.rpm
18:20:57     14/140 : curl-debuginfo-7.79.1-7.oe2203.x86_64.rpm
18:20:58     15/140 : curl-debugsource-7.79.1-7.oe2203.x86_64.rpm
18:20:59     16/140 : curl-help-7.79.1-7.oe2203.noarch.rpm
18:20:59     17/140 : bpftool-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:21:05     18/140 : containerd-1.2.0-303.oe2203.x86_64.rpm
18:21:07     19/140 : atune-1.0.0-4.oe2203.x86_64.rpm
18:21:08     20/140 : ebtables-2.0.11-5.oe2203.x86_64.rpm
18:21:08     21/140 : ebtables-debuginfo-2.0.11-5.oe2203.x86_64.rpm
18:21:08     22/140 : docker-engine-debugsource-18.09.0-306.oe2203.x86_64.rpm
18:21:08     23/140 : ebtables-debugsource-2.0.11-5.oe2203.x86_64.rpm
18:21:09     24/140 : ebtables-help-2.0.11-5.oe2203.x86_64.rpm
18:21:10     25/140 : gnupg2-2.2.32-2.oe2203.x86_64.rpm
18:21:11     26/140 : atune-engine-1.0.0-4.oe2203.x86_64.rpm
18:21:11     27/140 : gnupg2-debugsource-2.2.32-2.oe2203.x86_64.rpm
18:21:12     28/140 : gnupg2-help-2.2.32-2.oe2203.noarch.rpm
18:21:12     29/140 : gnupg2-debuginfo-2.2.32-2.oe2203.x86_64.rpm
18:21:13     30/140 : iSulad-debugsource-2.0.14-10.oe2203.x86_64.rpm
18:21:13     31/140 : initscripts-10.12-2.oe2203.x86_64.rpm
18:21:13     32/140 : iSulad-2.0.14-10.oe2203.x86_64.rpm
18:21:13     33/140 : initscripts-debuginfo-10.12-2.oe2203.x86_64.rpm
18:21:13     34/140 : initscripts-debugsource-10.12-2.oe2203.x86_64.rpm
18:21:14     35/140 : initscripts-service-10.12-2.oe2203.noarch.rpm
18:21:14     36/140 : iptables-1.8.7-8.oe2203.x86_64.rpm
18:21:14     37/140 : iptables-debugsource-1.8.7-8.oe2203.x86_64.rpm
18:21:14     38/140 : iptables-devel-1.8.7-8.oe2203.x86_64.rpm
18:21:15     39/140 : iptables-debuginfo-1.8.7-8.oe2203.x86_64.rpm
18:21:15     40/140 : iptables-help-1.8.7-8.oe2203.noarch.rpm
18:21:15     41/140 : iptables-libs-1.8.7-8.oe2203.x86_64.rpm
18:21:15     42/140 : iptables-nft-1.8.7-8.oe2203.x86_64.rpm
18:21:19     43/140 : iSulad-debuginfo-2.0.14-10.oe2203.x86_64.rpm
18:21:25     44/140 : docker-engine-debuginfo-18.09.0-306.oe2203.x86_64.rpm
18:21:44     45/140 : docker-engine-18.09.0-306.oe2203.x86_64.rpm
18:21:51     46/140 : kernel-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:21:55     47/140 : kernel-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:22:09     48/140 : kernel-devel-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:22:24     49/140 : kernel-devel-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:22:27     50/140 : kernel-headers-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:22:30     51/140 : kernel-headers-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:22:49     52/140 : kernel-debugsource-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:22:51     53/140 : kernel-debugsource-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:22:53     54/140 : kernel-tools-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:22:54     55/140 : kernel-tools-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:22:55     56/140 : kernel-tools-debuginfo-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:22:55     57/140 : kernel-tools-debuginfo-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:22:56     58/140 : kernel-tools-devel-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:22:57     59/140 : kernel-tools-devel-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:22:58     60/140 : lcr-2.0.8-1.oe2203.x86_64.rpm
18:22:59     61/140 : lcr-debuginfo-2.0.8-1.oe2203.x86_64.rpm
18:22:59     62/140 : lcr-debugsource-2.0.8-1.oe2203.x86_64.rpm
18:22:59     63/140 : lcr-devel-2.0.8-1.oe2203.x86_64.rpm
18:23:00     64/140 : libarchive-3.5.2-4.oe2203.x86_64.rpm
18:23:02     65/140 : libarchive-debuginfo-3.5.2-4.oe2203.x86_64.rpm
18:23:02     66/140 : libarchive-debugsource-3.5.2-4.oe2203.x86_64.rpm
18:23:03     67/140 : libarchive-devel-3.5.2-4.oe2203.x86_64.rpm
18:23:03     68/140 : libarchive-help-3.5.2-4.oe2203.noarch.rpm
18:23:03     69/140 : libcurl-7.79.1-7.oe2203.x86_64.rpm
18:23:04     70/140 : libcurl-devel-7.79.1-7.oe2203.x86_64.rpm
18:23:04     71/140 : libreswan-4.5-2.oe2203.x86_64.rpm
18:23:06     72/140 : libreswan-debuginfo-4.5-2.oe2203.x86_64.rpm
18:23:07     73/140 : libreswan-debugsource-4.5-2.oe2203.x86_64.rpm
18:23:07     74/140 : libreswan-help-4.5-2.oe2203.x86_64.rpm
18:23:08     75/140 : libtiff-4.3.0-16.oe2203.x86_64.rpm
18:23:08     76/140 : libtiff-debuginfo-4.3.0-16.oe2203.x86_64.rpm
18:23:09     77/140 : libtiff-debugsource-4.3.0-16.oe2203.x86_64.rpm
18:23:09     78/140 : libtiff-devel-4.3.0-16.oe2203.x86_64.rpm
18:23:10     79/140 : libtiff-help-4.3.0-16.oe2203.noarch.rpm
18:23:10     80/140 : libtiff-static-4.3.0-16.oe2203.x86_64.rpm
18:23:11     81/140 : libtiff-tools-4.3.0-16.oe2203.x86_64.rpm
18:23:11     82/140 : lxc-4.0.3-2022052501.oe2203.x86_64.rpm
18:23:12     83/140 : lxc-debuginfo-4.0.3-2022052501.oe2203.x86_64.rpm
18:23:13     84/140 : lxc-debugsource-4.0.3-2022052501.oe2203.x86_64.rpm
18:23:13     85/140 : lxc-devel-4.0.3-2022052501.oe2203.x86_64.rpm
18:23:15     86/140 : lxc-help-4.0.3-2022052501.oe2203.noarch.rpm
18:23:15     87/140 : lxc-libs-4.0.3-2022052501.oe2203.x86_64.rpm
18:23:16     88/140 : netconsole-service-10.12-2.oe2203.noarch.rpm
18:23:16     89/140 : network-scripts-10.12-2.oe2203.x86_64.rpm
18:23:16     90/140 : oec-hardware-1.1.0-1.oe2203.x86_64.rpm
18:23:17     91/140 : oec-hardware-server-1.1.0-1.oe2203.x86_64.rpm
18:23:17     92/140 : openssl-1.1.1m-6.oe2203.x86_64.rpm
18:23:20     93/140 : openssl-debuginfo-1.1.1m-6.oe2203.x86_64.rpm
18:23:22     94/140 : openssl-debugsource-1.1.1m-6.oe2203.x86_64.rpm
18:23:23     95/140 : openssl-devel-1.1.1m-6.oe2203.x86_64.rpm
18:23:28     96/140 : openssl-help-1.1.1m-6.oe2203.noarch.rpm
18:23:30     97/140 : openssl-libs-1.1.1m-6.oe2203.x86_64.rpm
18:23:31     98/140 : openssl-perl-1.1.1m-6.oe2203.x86_64.rpm
18:23:36     99/140 : perf-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:23:41     100/140 : kernel-source-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:23:42     101/140 : perf-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:23:44     102/140 : perf-debuginfo-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:23:45     103/140 : python3-perf-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:23:45     104/140 : python3-perf-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:23:46     105/140 : python3-perf-debuginfo-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:23:47     106/140 : python3-perf-debuginfo-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:23:48     107/140 : python3-rpm-4.17.0-9.oe2203.x86_64.rpm
18:23:48     108/140 : readonly-root-10.12-2.oe2203.noarch.rpm
18:23:49     109/140 : rpm-4.17.0-9.oe2203.x86_64.rpm
18:23:49     110/140 : rpm-build-4.17.0-9.oe2203.x86_64.rpm
18:23:50     111/140 : rpm-debuginfo-4.17.0-9.oe2203.x86_64.rpm
18:23:51     112/140 : rpm-debugsource-4.17.0-9.oe2203.x86_64.rpm
18:23:51     113/140 : rpm-devel-4.17.0-9.oe2203.x86_64.rpm
18:23:52     114/140 : rpm-help-4.17.0-9.oe2203.noarch.rpm
18:23:52     115/140 : perf-debuginfo-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:23:52     116/140 : rpm-libs-4.17.0-9.oe2203.x86_64.rpm
18:23:53     117/140 : rpm-plugin-systemd-inhibit-4.17.0-9.oe2203.x86_64.rpm
18:23:53     118/140 : selinux-policy-35.5-6.oe2203.noarch.rpm
18:23:53     119/140 : selinux-policy-help-35.5-6.oe2203.noarch.rpm
18:23:54     120/140 : selinux-policy-devel-35.5-6.oe2203.noarch.rpm
18:23:57     121/140 : selinux-policy-minimum-35.5-6.oe2203.noarch.rpm
18:23:57     122/140 : selinux-policy-sandbox-35.5-6.oe2203.noarch.rpm
18:24:01     123/140 : selinux-policy-targeted-35.5-6.oe2203.noarch.rpm
18:24:01     124/140 : swtpm-0.3.3-6.oe2203.x86_64.rpm
18:24:01     125/140 : swtpm-debuginfo-0.3.3-6.oe2203.x86_64.rpm
18:24:02     126/140 : swtpm-debugsource-0.3.3-6.oe2203.x86_64.rpm
18:24:02     127/140 : swtpm-devel-0.3.3-6.oe2203.x86_64.rpm
18:24:03     128/140 : swtpm-libs-0.3.3-6.oe2203.x86_64.rpm
18:24:03     129/140 : swtpm-tools-0.3.3-6.oe2203.x86_64.rpm
18:24:04     130/140 : selinux-policy-mls-35.5-6.oe2203.noarch.rpm
18:24:04     131/140 : vim-X11-8.2-48.oe2203.x86_64.rpm
18:24:11     132/140 : vim-debuginfo-8.2-48.oe2203.x86_64.rpm
18:24:13     133/140 : vim-debugsource-8.2-48.oe2203.x86_64.rpm
18:24:14     134/140 : vim-enhanced-8.2-48.oe2203.x86_64.rpm
18:24:14     135/140 : vim-filesystem-8.2-48.oe2203.noarch.rpm
18:24:15     136/140 : vim-minimal-8.2-48.oe2203.x86_64.rpm
18:24:17     137/140 : vim-common-8.2-48.oe2203.x86_64.rpm
18:24:42     138/140 : kernel-source-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:24:46     139/140 : kernel-debuginfo-5.10.0-60.43.0.72.oe2203.x86_64.rpm
18:25:41     140/140 : kernel-debuginfo-5.10.0-60.42.0.71.oe2203.x86_64.rpm
18:25:41 
18:25:41   Importing packages to DB:
18:29:25   Package batch #5 of 7 completed...
18:29:48   Package batch #6 of 7 completed...
18:30:17   Package batch #1 of 7 completed...
18:30:29   Package batch #4 of 7 completed...
18:30:34   Package batch #2 of 7 completed...
18:30:38   Package batch #3 of 7 completed...
18:30:59   Package batch #7 of 7 completed...
18:30:59 
18:30:59   Linking packages to the channel.
18:31:01     140 packages linked
18:31:01 
18:31:01   Patches in repo: 124.
18:31:01 Unexpected error: <class 'TypeError'>
18:31:01 Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 620, in sync
    self.import_updates(plugin)
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 763, in import_updates
    processed_updates_count = self.upload_updates(notices)
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 966, in upload_updates
    notice = self.fix_notice(notice)
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 1541, in fix_notice
    if "." in notice['version']:
TypeError: argument of type 'NoneType' is not iterable

18:31:02   Regenerating bootstrap repositories.
Generating bootstrap repos for all available products which had changes.

Creating bootstrap repo for openeuler22.03-aarch64-uyuni
copy 'venv-salt-minion-3004-4.2.uyuni.aarch64'
Directory walk started
Directory walk done - 2 packages
Temporary output repo path: /srv/www/htdocs/pub/repositories/openEuler/22.03/bootstrap.tmp/.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Pool finished
18:31:07 Total time: 0:10:47

```

**Comment by vzhestkov** [comment]
@raulillo82 I'm closing the issue as the fix was merged. It will not be available in `2022.08`, but on the next release only
